packages/storage/src/implementation/observer.ts

```diff
@@ -14,26 +14,67 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import * as type from './type';
+import { isFunction } from './type';
 import { FirebaseStorageError } from './error';
 
+/**
+ * @public
+ * Function that is called once for each value in a stream of values.
+ */
 export type NextFn<T> = (value: T) => void;
+
+/**
+ * @public
+ * A function that is called with a `FirebaseStorageError`
+ * if the event stream ends due to an error.
+ */
 export type ErrorFn = (error: FirebaseStorageError) => void;
+
+/**
+ * @public
+ * A function that is called if the event stream ends normally.
+ */
 export type CompleteFn = () => void;
+
+/**
+ * @public
+ * Unsubscribes from a stream.
+ */
 export type Unsubscribe = () => void;
 
+/**
+ * @public
+ * An observer that wraps any errors in `FirebaseStorageError`.
```

Comment:
I changed exp/index to export the public `Observer` type which should have the public comment. This is now an internal comment just for us to explain the only reason we need a special StorageObserver is that it wraps errors in `FirebaseStorageError` instead of a generic `Error`.

Reply:
I still don't understand it. What does it mean to have observer wrap errors in `FirebaseStorageError`?

Reply:
I tried to update the comment to be more clear.

General Observer type that you can get from @firebase/util is roughly
```
Observer<T> {
  next?: (value: T) => void;
  error?: (error: Error) => void;
  complete?: () => void;
}
```
In a `StorageObserver` type the error function parameter is a `FirebaseStorageError` instead of `Error`.

Reply:
Ah I see